azimuth/startup.py

```diff
@@ -146,16 +183,22 @@ def make_startup_tasks(
     dms = {k: v for k, v in dataset_split_managers.items() if v is not None}
     tasks: Dict[DatasetSplitName, DaskModule] = {}
 
-    for dataset_split_name, dm in dms.items():
+    if DatasetSplitName.all not in dataset_split_names:
+        available_datasets_splits = list(set(dataset_split_names) & set(dms))
+    else:
+        available_datasets_splits = [DatasetSplitName.all]
+
+    for dataset_split_name in available_datasets_splits:
         _, maybe_task = task_manager.get_task(
             task_name=supported_module,
             dataset_split_name=dataset_split_name,
             dependencies=dependencies,
             mod_options=ModuleOptions(pipeline_index=pipeline_index, **mod_options),
         )
         task = assert_not_none(maybe_task)
-        if task.future is not None:
-            # If we had to launch something, register the callback.
+        task_launched = task.future is not None
+        if task_launched:
+            dm = dms[dataset_split_name] if dataset_split_name != DatasetSplitName.all else dms
```

Comment:
... When I use an `else`, I avoid inverting/negating the condition to keep it simple. Otherwise, this is like a double negation. In this case, I would use `==` instead of `!=` and flip.
```suggestion
            dm = dms if dataset_split_name == DatasetSplitName.all else dms[dataset_split_name]
```